### PR TITLE
Implemented the SaveProgramBanList method. Closed #17

### LINKED ISSUE #17: Implement the SaveProgramBanList method
Implement the SaveProgramBanList method in the ProgramBanListService class. Add different saving options: to file and to server.

### PR DESCRIPTION
**Pull request by Andrey31ts**

### CODE REVIEW

**Review by cany8042 — DISMISSED** [review]

File: user-monitoring-gui/Services/ProgramBanListService.cs [hunk 1]
```diff
@@ -1,12 +1,44 @@
-﻿using user_monitoring_gui.Models;
+﻿using System.Text;
+using user_monitoring_gui.Models;
 using user_monitoring_gui.Services.Interfaces;
 
 namespace user_monitoring_gui.Services
 {
-    public class ProgramBanListService : IProgramBanListService
+    public class ProgramBanListService:IProgramBanListService
```
Comment (TopAcademy2022) [inline user-monitoring-gui/Services/ProgramBanListService.cs]: Add spaces before ":" symbol.

File: user-monitoring-gui/Services/ProgramBanListService.cs [hunk 2]
```diff
@@ -1,12 +1,44 @@
...
     {
-        public bool Save(ProgramBanList programBanList)
+
```
Comment (TopAcademy2022) [inline user-monitoring-gui/Services/ProgramBanListService.cs:10]: Remove this line.

File: user-monitoring-gui/Services/ProgramBanListService.cs [hunk 3]
```diff
@@ -1,12 +1,44 @@
...
+   private IServerRequest _serverReqwest;
```
Comment (TopAcademy2022) [inline user-monitoring-gui/Services/ProgramBanListService.cs]: Add tabs.

File: user-monitoring-gui/Services/ProgramBanListService.cs [hunk 4]
```diff
@@ -1,12 +1,44 @@
...
+
+        public ProgramBanListService(IServerRequest serverReqwest)
+        {
+            this._serverReqwest = serverReqwest;
+        }
+        public bool Save(ProgramBanList programBanList, DataStorageArea dataStorageArea)
```
Comment (TopAcademy2022) [inline user-monitoring-gui/Services/ProgramBanListService.cs:16]: Add space.

File: user-monitoring-gui/Services/ProgramBanListService.cs
Comment (TopAcademy2022) [inline user-monitoring-gui/Services/ProgramBanListService.cs:1]: Update IProgramBanListService.

**Review by TopAcademy2022 — CHANGES_REQUESTED** [review]

**Review by TopAcademy2022 — DISMISSED** [review]

File: user-monitoring-gui/Services/ProgramBanListService.cs [hunk 5]
```diff
@@ -1,16 +1,48 @@
...
+using user_monitoring_gui.Models;
 using user_monitoring_gui.Services.Interfaces;
 
 namespace user_monitoring_gui.Services
 {
     public class ProgramBanListService : IProgramBanListService
     {
-        public bool Save(ProgramBanList programBanList)
+
+	private IServerRequest _serverReqwest;
```
Comment (HecuseS) [inline user-monitoring-gui/Services/ProgramBanListService.cs]: replace the word "Reqwest" with the correct spelling - Request

File: user-monitoring-gui/Services/ProgramBanListService.cs [hunk 6]
```diff
@@ -1,16 +1,48 @@
...
+                    return true;
+
+                case DataStorageArea.SERVER:
+                    foreach (var item in bytes)
+                    {
+                        this._serverReqwest.SendData(item);
+                    }
+                    return true;
+            }
+
```
Comment (HecuseS) [inline user-monitoring-gui/Services/ProgramBanListService.cs]: Delite line

**Review by HecuseS — CHANGES_REQUESTED** [review]

File: user-monitoring-gui/Services/ProgramBanListService.cs [hunk 7]
```diff
@@ -1,16 +1,47 @@
...
+                case DataStorageArea.SERVER:
+                    foreach (var item in bytes)
+                    {
+                        this._serverRequest.SendData(item);
+                    }
+                    return true;
+            }
+            return false;
+        }
+   public bool Load()
```
Comment (TopAcademy2022) [inline user-monitoring-gui/Services/ProgramBanListService.cs]: Add space.

**Review by TopAcademy2022 — CHANGES_REQUESTED** [review]

**Review by HecuseS — DISMISSED** [review]

**Review by TopAcademy2022 — DISMISSED** [review]

**Review by London3149-train — DISMISSED** [review]

**Review by cany8042 — DISMISSED** [review]

**Review by HecuseS — DISMISSED** [review]

**Review by xs887z — DISMISSED** [review]

**Review by SugarSalt9 — DISMISSED** [review]

**Review by Mi-stik31 — DISMISSED** [review]

**Review by xs887z — APPROVED** [review]

**Review by xs887z — DISMISSED** [review]

**Review by HecuseS — APPROVED** [review]

**Review by TopAcademy2022 — APPROVED** [review]

**Review by xs887z — APPROVED** [review]

**Review by Mi-stik31 — APPROVED** [review]

**Review by cany8042 — APPROVED** [review]

**Review by London3149-train — APPROVED** [review]

**Review by SugarSalt9 — APPROVED** [review]

**Pull request merged**